src/arithmeticnode.cpp

```diff
@@ -0,0 +1,97 @@
+// arithmeticnode.cpp
+#include "arithmeticnode.h"
+#include <QtGui/QDoubleValidator>
+
+ArithmeticNode::ArithmeticNode(const QString& operationName) : m_operationName(operationName) {
+    m_bufferData = std::make_shared<BufferData>();
+    m_bufferData->setAll(0.0f);
+}
+
+QString ArithmeticNode::caption() const {
+    return m_operationName + " Node";
+}
+
+QString ArithmeticNode::name() const {
+    return m_operationName + "Node";
+}
+
+bool ArithmeticNode::captionVisible() const {
+    return true;
+}
+
+unsigned int ArithmeticNode::nPorts(QtNodes::PortType portType) const {
+    if (portType == QtNodes::PortType::Out) return 1;
+    return 2;  // Two input ports
+}
+
+QtNodes::NodeDataType ArithmeticNode::dataType(QtNodes::PortType, QtNodes::PortIndex) const {
+    return BufferData{}.type();
+}
+
+void ArithmeticNode::setInData(std::shared_ptr<QtNodes::NodeData> data, QtNodes::PortIndex portIndex) {
+    if (auto inputData = std::dynamic_pointer_cast<BufferData>(data)) {
+        m_bufferData->m_buffer[portIndex] = inputData->m_buffer[0];
+
+        // Perform the operation
+        float result = performOperation(m_bufferData->m_buffer[0], m_bufferData->m_buffer[1]);
+

```

Comment:
this will just perform the operation on the first two elements of the same buffer and set every element in the output buffer to that value. The intended behavior is to perform the operation between elements with the same indices in each input buffer and store the result in that index in the output buffer